include/libtorrent/aux_/vector.hpp

```diff
@@ -70,7 +70,22 @@ namespace libtorrent { namespace aux {
 		}
 
 		IndexType end_index() const
-		{ return IndexType(static_cast<underlying_index>(this->size())); }
+		{
+			TORRENT_ASSERT(this->size() <= std::size_t(std::numeric_limits<underlying_index>::max()));
+			return IndexType(static_cast<underlying_index>(this->size()));
+		}
+
```

Comment:
if you add a line: ``using base::resize;`` here, you'll pull in the overloads from the base class too